lldb/source/Plugins/SymbolFile/DWARF/UniqueDWARFASTType.cpp

```diff
@@ -13,12 +13,18 @@
 using namespace lldb_private::dwarf;
 using namespace lldb_private::plugin::dwarf;
 
+static bool IsStructOrClassTag(llvm::dwarf::Tag Tag) {
```

Comment:
I think we have this function somewhere already. Might be worth checking (possibly even in the llvm headers?)

Reply:
Ah yes in `SymbolFileDWARF::FindDefinitionTypeForDWARFDeclContext` we have the same kind of check. Maybe one could consolidate those in one API? But might not be worth the churn, so feel free to ignore

Reply:
It's here: https://github.com/llvm/llvm-project/blob/ed35a92c404650b15a79ff38bcaff41de176cb78/lldb/source/Plugins/SymbolFile/DWARF/SymbolFileDWARF.cpp#L136, maybe move it to namespace `lldb_private::plugin::dwarf`?

Reply:
Are there more tag equality checks around LLDB that could benefit from re-using the following check:
```
udt.m_die.Tag() == die.Tag() || (IsStructOrClassTag(udt.m_die.Tag()) &&
                                         IsStructOrClassTag(die.Tag()))
```

There's at least two now. Not sure where we'd put such an API. Perhaps @felipepiovezan or @adrian-prantl have some input on this.

Reply:
Yeah, we can probably do it in a different change.

Reply:
There's also
```
/// Returns true if `tag` is a class_type of structure_type tag.  
static bool IsClassOrStruct(dw_tag_t tag) {                       
  return tag == DW_TAG_class_type || tag == DW_TAG_structure_type;
}                                                                 
```
In `AppleDWARFIndex` FYI

Reply:
Yeah, we definitely need to clean all of these up.
There might be one inside llvm/ too..